text/2020-04-23-rawkv-txnkv-mode-switching.md

```diff
@@ -0,0 +1,63 @@
+# Title
+
+## Summary
+
+This RFC proposes to forbid users using RawKV API on the TiKV cluster by default,
+and provide a new API command to let users enable it.
+
+## Motivation
+
+Some painful incidents happened before and still happening,
+because of the mixed-using of RawKV API and TransactionKV API,
+some key-values were overwritten by RawKV API,
+which originally written by TransactionKV API,
+then led to data corruption.
+
+We should add more warnings about this in our documents, but it's not enough,
+documents could easily be ignored.
+
+Providing a switching API could prevent people from misuse without noticing,
+when users need to step into the danger zone,
+they need to explicitly open the guarded switch.
+And they could shut down the switch anytime to gain more safety.
+
+## Detailed design
+
+This `mode` flag should be persisted,
```

Comment:
where does this get persisted? is it a special key in the underlying engine, or in a separate file?

Reply:
If we use dynamic config as you mention above, then it will be persisted along with other dynamic config entries (in PD)